launch/launch/logging/__init__.py

```diff
@@ -300,6 +305,14 @@ def get_logger(name=None):
     return logger
 
 
+@expose_substitution('log_dir')
+def _log_dir(data: Iterable[SomeSubstitutionsType]):
+    if len(data) > 0:
+        raise ValueError('log_dir substitution does not expect any arguments')
+    # Parser expects callable
+    return lambda: TextSubstitution(text=launch_config.log_dir), {}
```

Comment:
@jacobperron may I suggest:
```suggestion
    return TextSubstitution, {'text': launch_config.log_dir}
```

Reply:
Thanks, the lambda was vestigial. Applied suggestion in 2a71fc2